neanderthal/src/main/kotlin/au/com/outware/neanderthal/data/repository/VariantSharedPreferencesRepository.kt

```diff
@@ -52,9 +52,14 @@ class VariantSharedPreferencesRepository(val klass: Class<out Any>,
             for (variant in baseVariants) {
                 editor.putString(variant.key, gson.toJson(variant.value))
             }
+
             editor.putStringSet(VARIANT_STRUCTURE, structure)
-            editor.putString(CURRENT_VARIANT, defaultVariant)
-            editor.putString(DEFAULT_VARIANT, defaultVariant)
+
+            // Set the default/current variant to the one supplied, or the first if the supplied default doesnt exist
+            val currentVariant = if (baseVariants.containsKey(defaultVariant)) defaultVariant else baseVariants.keys.first()
```

Comment:
thinking further about this, if we dont specify a default variant then behaviour for production is undefined. I've changed the logic to instead abuse the dev for supplying an incorrect default